File: 1482-how-many-numbers-are-smaller-than-the-current-number/1482-how-many-numbers-are-smaller-than-the-current-number.cpp
class Solution {
public:
    vector<int> smallerNumbersThanCurrent(vector<int>& nums) {
        vector<int> Final; 
        int left = 0; 
        int right = 0;
        int count = 0; 

        while(right <= nums.size() && left < nums.size()){
            if(right == nums.size()){
                Final.push_back(count); 
                count = 0; 
                right = 0; 
                left++; 
                continue; 
            }
            if(right != left){
                if(nums[right] < nums[left]){
                    count++; 
                }
            }
            right++; 
        } 

        return Final; 

        
        
    }
};

File: 1482-how-many-numbers-are-smaller-than-the-current-number/1482-how-many-numbers-are-smaller-than-the-current-number.py
class Solution:
    def smallerNumbersThanCurrent(self, nums: List[int]) -> List[int]:
        compare = sorted(nums)
        mapp = defaultdict(int)
        for i in range(len(compare)): 
            if compare[i] not in mapp: 
                mapp[compare[i]] = i
        
        res = []
        for i in range(len(nums)): 
            res.append(mapp[nums[i]])

        return res 

